posthog/hogql/test/test_expr_parser.py

```diff
@@ -0,0 +1,164 @@
+from typing import Any, Dict
+
+from posthog.hogql.expr_parser import ExprParserContext, translate_hql
+from posthog.test.base import APIBaseTest, ClickhouseTestMixin
+
+
+class TestExprParser(APIBaseTest, ClickhouseTestMixin):
+    def test_hogql_literals(self):
+        self.assertEqual(translate_hql("1 + 2"), "plus(1, 2)")
+        self.assertEqual(translate_hql("-1 + 2"), "plus(-1, 2)")
+        self.assertEqual(translate_hql("-1 - 2 / (3 + 4)"), "minus(-1, divide(2, plus(3, 4)))")
+        self.assertEqual(translate_hql("1.0 * 2.66"), "multiply(1.0, 2.66)")
+        self.assertEqual(translate_hql("1.0 % 2.66"), "modulo(1.0, 2.66)")
+        self.assertEqual(translate_hql("'string'"), "'string'")
+        self.assertEqual(translate_hql('"string"'), "'string'")
+
+    def test_hogql_fields_and_properties(self):
+        self.assertEqual(
+            translate_hql("properties.bla"),
+            "replaceRegexpAll(JSONExtractRaw(properties, 'bla'), '^\"|\"$', '')",
+        )
+        self.assertEqual(
+            translate_hql("properties['bla']"),
+            "replaceRegexpAll(JSONExtractRaw(properties, 'bla'), '^\"|\"$', '')",
+        )
+        self.assertEqual(
+            translate_hql('properties["bla"]'),
+            "replaceRegexpAll(JSONExtractRaw(properties, 'bla'), '^\"|\"$', '')",
+        )
+        self.assertEqual(
+            translate_hql("properties.$bla"),
+            "replaceRegexpAll(JSONExtractRaw(properties, '$bla'), '^\"|\"$', '')",
+        )
+        self.assertEqual(
+            translate_hql("person.properties.bla"),
+            "replaceRegexpAll(JSONExtractRaw(person_properties, 'bla'), '^\"|\"$', '')",
+        )
+        self.assertEqual(translate_hql("uuid"), "uuid")
+        self.assertEqual(translate_hql("event"), "event")
+        self.assertEqual(translate_hql("timestamp"), "timestamp")
+        self.assertEqual(translate_hql("distinct_id"), "distinct_id")
+        self.assertEqual(translate_hql("person_id"), "person_id")
+        self.assertEqual(translate_hql("person.created_at"), "person_created_at")
+
+    def test_hogql_materialized_fields_and_properties(self):
```

Comment:
Test for materialized person properties?

Reply:
Nice! That fixed a bug.